cli/text2embeddings.py

```diff
@@ -165,17 +165,20 @@ def main(
             f"The following languages have been requested for encoding but are not supported by the encoder: {unsupported_languages}. Only the following languages will be encoded: {config.ENCODER_SUPPORTED_LANGUAGES}."
         )
 
+    # Encode documents either with one language where the lanugage is in the target languages, or with no language and no content type. This assumes that the document name and description are in English.
     tasks = [
         task
         for task in tasks
-        if task.languages
-        and (len(task.languages) == 1)
-        and (
-            task.languages[0]
-            in config.ENCODER_SUPPORTED_LANGUAGES.union(config.TARGET_LANGUAGES)
+        if (
+            task.languages
+            and (len(task.languages) == 1)
+            and (
+                task.languages[0]
+                in config.ENCODER_SUPPORTED_LANGUAGES.union(config.TARGET_LANGUAGES)
+            )
         )
+        or (not task.languages and task.html_data is None and task.pdf_data is None)
```

Comment:
this looks like it will get us through, but after the dust has settled we really need to sit down & document the pre/post-conditions for each step in the pipeline!